core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```diff
@@ -125,27 +127,19 @@ oppia.factory('AnswerClassificationService', [
           return deferred.promise;
         }
 
-        if (result.outcome === defaultOutcome &&
-            INTERACTION_SPECS[oldState.interaction.id]
-              .is_string_classifier_trainable &&
+        var ruleBasedOutcomeIsDefault = (result.outcome == defaultOutcome);
+        var interactionIsTrainable = INTERACTION_SPECS[
+          oldState.interaction.id].is_string_classifier_trainable;
```

Comment:
Why is this `is_string_classifier_trainable`? Should not we change it to more general, since it will be present for all trainable interactions.